client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -28,523 +26,383 @@ import {
 	WEB_PAYMENT_BASIC_CARD_METHOD,
 	WEB_PAYMENT_APPLE_PAY_METHOD,
 } from 'lib/web-payment';
-import wpcom from 'lib/wp';
-import { newCardPayment } from 'lib/store-transactions';
-import { setPayment } from 'lib/upgrades/actions';
-import {
-	BEFORE_SUBMIT,
-	INPUT_VALIDATION,
-	RECEIVED_PAYMENT_KEY_RESPONSE,
-	RECEIVED_WPCOM_RESPONSE,
-	REDIRECTING_FOR_AUTHORIZATION,
-	SUBMITTING_PAYMENT_KEY_REQUEST,
-	SUBMITTING_WPCOM_REQUEST,
-} from 'lib/store-transactions/step-types';
-import notices from 'notices';
+import { webPayment } from 'lib/store-transactions';
+import { setPayment, setStripeObject } from 'lib/upgrades/actions';
+import { setTaxCountryCode, setTaxPostalCode } from 'lib/upgrades/actions/cart';
 import CartToggle from './cart-toggle';
 import CheckoutTerms from './checkout-terms';
 import PaymentChatButton from './payment-chat-button';
 import RecentRenewals from './recent-renewals';
 import SubscriptionText from './subscription-text';
-import { setTaxCountryCode, setTaxPostalCode } from 'lib/upgrades/actions/cart';
+import { withStripe } from 'lib/stripe';
+import { useDebounce } from 'blocks/credit-card-form/helpers';
 
-const debug = debugFactory( 'calypso:checkout:payment:apple-pay' );
+const debug = debugFactory( 'calypso:checkout:payment:web-payment-box' );
 
-/**
- * Supported card types.
- */
-const SUPPORTED_NETWORKS = [ 'visa', 'masterCard', 'discover', 'amex', 'jcb', 'chinaUnionPay' ];
-const APPLE_PAY_MERCHANT_IDENTIFIER = config( 'apple_pay_merchant_id' );
 const PAYMENT_REQUEST_OPTIONS = {
 	requestPayerName: true,
```

Comment:
I don't understand why, but this is apparently forcing us to collect not just the name but also the entire billing address.

Stripe even documents that: https://stripe.com/docs/stripe-js/reference#stripe-payment-request ("We highly recommend you collect at least one of name, email, or phone as this also results in collection of billing address for Apple Pay")

With regular Apple Pay (our current implementation) it just collects the name, which is all we need.  I'm pretty sure it only requires the billing address if you directly ask for it.

This change in behavior is kind of bad because it means we're requiring Apple Pay users to type in lots of extra information that we don't even require from credit card users (although at least hopefully many of them have typed it in and saved it already)... maybe we need to reach out to Stripe to understand what is going on here.

Reply:
And another side effect of that (I think) is that it results in the postal code that the user typed before opening Apple Pay being overwritten (in the data saved server-side) with the one that came from the billing address in Apple Pay -- in the case where they differ.

I think in general we have been trying to consistently save the one the user typed since that is the one used to calculate taxes on checkout.

Reply:
🤔 Thanks. I need to take a deeper look at this. I thought we were sending the user-supplied postal code and ignoring the other billing data when it's submitted but I certainly could be wrong.

Reply:
> This change in behavior is kind of bad because it means we're requiring Apple Pay users to type in lots of extra information

@DavidRothstein I'd like to know more about what you discovered about this since I don't see that I have to enter any information at all during checkout with this option. I just choose a card from the list, and when I entered the card in the first place (using my phone, which AFAIK is the only way to do it), it didn't ask for the billing address at all (it might get it from the credit card processor, I suppose). Maybe there's different flows depending on the card?

> the postal code that the user typed before opening Apple Pay being overwritten (in the data saved server-side) with the one that came from the billing address in Apple Pay

That could be... I haven't checked the server side code yet, but I'm not so sure because on the client we grab the `payerName` and `paymentMethod` and then (in `completePaymentMethodTransaction()`) combine them with the user-selected `postalCode` and `countryCode` to send to the TransactionStore and thence to the server. Was there somewhere specific you saw this happen or suspected it of happening? (There is an unclear object merge right before the call to `completePaymentMethodTransaction` which could in theory overwrite the user supplied values that I'm going to make more explicit, but I checked and the path to the postal data is quite different from our key.)

Reply:
>I just choose a card from the list, and when I entered the card in the first place (using my phone, which AFAIK is the only way to do it), it didn't ask for the billing address at all (it might get it from the credit card processor, I suppose). Maybe there's different flows depending on the card?

Hm, I added my card a long time ago, but I'm pretty sure I typed in my billing address by hand at the time.  (I didn't need it for anything, but I seem to remember that the field was empty and I filled it out while poking around the Apple Wallet.)  I wonder what other people experience?

To test it with this pull request, I just deleted the billing address associated with the card in my Apple Wallet and then confirmed that it forced me to type a new one in before I could purchase on WordPress.com.

>>the postal code that the user typed before opening Apple Pay being overwritten (in the data saved server-side) with the one that came from the billing address in Apple Pay
>
>Was there somewhere specific you saw this happen or suspected it of happening?

You can reproduce it with this pull request by typing a ZIP code in the WordPress.com form field that's different from the card's ZIP code in Apple Pay, then purchasing. If you look at the card record server-side (on WordPress.com), it will have the billing ZIP code from Apple Pay, not the one you typed in.

I noticed that if you go into the Stripe dashboard (https://dashboard.stripe.com) and look at the record there, they have the entire billing address from Apple Pay saved also.  So I think what might be happening here is that it's not anything in the JavaScript doing it; rather, Stripe gets the billing address directly from Apple Pay and saves it with the payment method, and then later the WordPress.com code on the server updates its own records with the data from Stripe (overwriting whatever it had for the ZIP code before).

Reply:
I think we could do that if we have to, but it basically has the opposite problem -- everyone will have to type their name in (rather than letting Apple Pay provide it to us automatically).

Maybe a good first step would be to talk with Stripe about why their API behaves this way and figure out f there's a way to get around it...

Reply:
I can confirm the zip code issue is still present - we save the user input country, and the Apple Pay billing address zip in our stored details. I'll take a look.

Reply:
Thinking about it again - we do want to store the proper zip code because it is used as a signal for card verification (especially in the US), and missing it might harm reuse/renewals of the card. I guess we'd want to set the card billing country as well. 

In terms of UX, the UI for these payments is going to change with our expected redesign, so when the steps will be seperate, it will be less of a problem.

Reply:
We've fixed this in the backend - for now, we'll update the PM on Stripe with the address (zip/country) the user has specifically entered in the form.